posts/a-beginners-guide-to-influxdb.md

```diff
@@ -0,0 +1,109 @@
+
+The last and the most important component of the line protocol is **time**, which is a **timestamp** in the RFC3339 UTC format. The timestamp precision can be in second, millisecond, microsecond or nanosecond. 
+
+
+# Writing Data to InfluxDB
+Data can be written to InfluxDB using command line interface using InfluxDB API or using client libraries. A simple curl, command line tool can be used to create and write to a database. InfluxDB provides client libraries for most of the famous languages like Java, PHP, Python, C++, Go etc. I have used Python influxdb-client and below is an example of how to write the data using python influxdb-client.
+
+ - Start with importing the required modules in your program
+
+    ``` python
+    from influxdb_client import InfluxDBClient, Point, WritePrecision
+    from influxdb_client.client.write_api import SYNCHRONOUS 
+    ```
+
+ - Then create an InfluxDB client (mention your url, token and org as below) and a writer object
+
+    ```python
+    url = "<my-url>"
+    token = "<my-token>"
+    client = InfluxDBClient(url=url, token=token, org=org)
+    write_api = client.write_api(write_options=SYNCHRONOUS)
+    ```
+
+ - Use the module ‘Point’ that you have imported from influxes_client to convert your data to point, or alternatively create a point yourself
+
+    ```python
+    # Using built-in functionality to create Point
+    p = Point("my_measurement").tag("location", "location1").field("temperature", 25).time(datetime.utcnow(),WritePrecision.NS)
+    
+    # Creating the point manually
+    measurement = "my_measurement"
+    tag_set = "location=location1"
+    field_set = "temperature=25"
+    date_in_nano = str(int(datetime.datetime.utcnow().timestamp() * 1000000000))
+    p = f"{measurement},{tag_set} {field_set} {date_in_nano}" 
+    ```
+ - Finally, write the point to a database. 
+
+    ```python
+    org = "<my-org>"
+    bucket = "<my-bucket>"
+    write_api.write(bucket, org, p) 
+    ```
+
+Here the bucket is a location where the data is stored. It has a retention policy and org attached to it. Retention policy consists of duration, replication factor and shard group duration. The default values for these are infinite duration, one and seven days respectively.
```

Comment:
Retention policy consists of duration, replication factor, and shard group duration.